engine/src/models/world.rs

```diff
@@ -41,8 +43,12 @@ impl World {
         }
     }
 
-    pub fn get_entity(&self, entity_id: i32) -> Option<Entity> {
-        self.entities.get(&entity_id).cloned()
+    pub fn get_entity(&self, entity_id: i32) -> EngineResult<Entity> {
```

Comment:
I can roll back to what I had: it would return an `Option`, and then the action layer can convert it to the result. It makes sense. In the "model world", not having an entity isn't a problem by itself. It's only a problem when an action wants to operate on it and it doesn't exist.